### PR TITLE
Use pundit policy to authorize actions on decks

### PR DESCRIPTION
**Pull request by ClaytonPassmore**
- Added Pundit gem
- Changed name of method that authorizes requests from `authorize` to `authorize_request`
- Added deck policy

### CODE REVIEW

**Comment by wcurtis** [comment]
@ClaytonPassmore On it this evening!

File: app/policies/deck_policy.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  end
+
+  def destroy?
+    # Deck must be owned by the user
+    record.user_id == user.id
+  end
+
+  def rate?
+    # Deck must be owned by the user or public
+    record.user_id == user.id or record.public
```
Comment (wcurtis) [inline app/policies/deck_policy.rb]: Nice use with custom actions 👌

File: app/controllers/decks_controller.rb [hunk 2]
```diff
@@ -10,12 +10,9 @@ def index
...
   # GET /decks/:id
   def show
+    authorize @deck
     # Render the deck and all of its cards.
-    if @deck.user_id == @user.id or @deck.public
-      render json: @deck, :include => {:cards => {:only => [:id, :front, :back]}}, status: :ok
-    else
-      render status: :not_found
-    end
+    render json: @deck, :include => {:cards => {:only => [:id, :front, :back]}}, status: :ok
```
Comment (wcurtis) [inline app/controllers/decks_controller.rb]: Nice use of include to limit attributes. [Active Model Serializers](https://github.com/rails-api/active_model_serializers) are the object oriented way of doing this. Similar idea as Pundit where this functionality is defined in a serliaizer object.

If you installed that gem you'd just add a file to define the attributes for the api - it's pretty sick.

In this case you'd setup a file called
```
app/serializers/decks_serializer.rb
```

Comment (ClaytonPassmore) [inline app/controllers/decks_controller.rb]: Ok cool, good to know!

**Pull request merged**